src/Solve.jl

```diff
@@ -310,6 +310,21 @@ function kernel(A::MatElem; side::Symbol = :right)
   return K
 end
 
+function kernel(A::MatElem{<:RingElement}; side::Symbol = :right)
+  check_option(side, [:right, :left], "side")
+
+  R, U = hnf_with_transform(lazy_transpose(A))
+  if side === :right
+    R, U = hnf_with_transform(lazy_transpose(A))
```

Comment:
```suggestion
  if side === :right
    R, U = hnf_with_transform(lazy_transpose(A))
```
this looks duplicate